Adding states for Benin(BJ) country 🇧🇯

manutheblacker opened the pull request:
I've added some states for the Benin country
 
### All Submissions:
* [x]  Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerceblob/master/.github/CONTRIBUTING.md)?
* [x]  Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x]  Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
 
### Changes proposed in this Pull Request:
These changes add all states for Benin country to the list of WooCommerce Countries. It give the ability to the customers, to pick and select the right states while he is buying on a WooCommerce store.
 
### How to test the changes in this Pull Request:
1. Pick a product on your store
2. On the checkout page select Benin as country
3. If you have selected Benin as country, you will need to pick the state name you want to pick and then proceed the order.
4. Go to the WooCommerce order for a Benin customer and you'll see the state name displayed into the order

### Changelog entry
> Adding states for Benin country 🇧🇯

manutheblacker committed:
adding states for benin country

I've added some states from the country : Benin

manutheblacker committed:
Update states.php

formating based on phpcs standard

claudiosanches commented on i18n/states.php:
```diff
@@ -172,6 +172,85 @@
 	'BH' => array(),
 	'BI' => array(),
+	'BJ' => array( // Benin states.
+		'BJ-01' => __( 'ABOMEY', 'woocommerce' ),
```
Those state names doesn't seems to match the list: https://github.com/unicode-org/cldr/blob/master/common/subdivisions/en.xml#L522-L533

Also it's all on uppercase, it need to be fixed.

manutheblacker replied:
Okay i'll apply the changes and commit it

manutheblacker replied:
I have applied changes as mentionned, does it match now?

manutheblacker replied:
@claudiosanches

manutheblacker committed:
Update states.php

updating by the right states name

manutheblacker committed:
Update states.php

rrennick reviewed (approved):
@manutheblacker Thanks for the help.

rrennick commented on i18n/states.php:
```diff
@@ -172,6 +172,20 @@
+		'BJ-03' => __( 'Atlantique', 'woocommerce' ),
+		'BJ-04' => __( 'Borgou', 'woocommerce' ),
+		'BJ-05' => __( 'Collines', 'woocommerce' ),
+		'BJ-06' => __( 'Couffo', 'woocommerce' ),
```
`Kouffo`.

claudiosanches replied:
```suggestion
		'KO' => __( 'Kouffo', 'woocommerce' ),
```

rrennick commented on i18n/states.php:
```diff
@@ -172,6 +172,20 @@
 	'BI' => array(),
+	'BJ' => array( // Benin states.
+		'BJ-01' => __( 'Alibori', 'woocommerce' ),
+		'BJ-02' => __( 'Atacora', 'woocommerce' ),
```
Noting for second reviewer that this is the French spelling for the state.

manutheblacker replied:
Yes, you got a point. This is the French spelling for this state because it is a French state.

claudiosanches commented on i18n/states.php:
```diff
@@ -172,6 +172,20 @@
 	'BH' => array(),
 	'BI' => array(),
+	'BJ' => array( // Benin states.
+		'BJ-01' => __( 'Alibori', 'woocommerce' ),
```
```suggestion
		'AL' => __( 'Alibori', 'woocommerce' ),
```

claudiosanches commented on i18n/states.php:
```diff
@@ -172,6 +172,20 @@
 	'BI' => array(),
+	'BJ' => array( // Benin states.
+		'BJ-01' => __( 'Alibori', 'woocommerce' ),
+		'BJ-02' => __( 'Atacora', 'woocommerce' ),
```
```suggestion
		'AK' => __( 'Atakora', 'woocommerce' ),
```

I think we should keep what the CLDR uses, it can be translated in other languages too.

claudiosanches commented on i18n/states.php:
```diff
@@ -172,6 +172,20 @@
+	'BJ' => array( // Benin states.
+		'BJ-01' => __( 'Alibori', 'woocommerce' ),
+		'BJ-02' => __( 'Atacora', 'woocommerce' ),
+		'BJ-03' => __( 'Atlantique', 'woocommerce' ),
```
```suggestion
		'AQ' => __( 'Atlantique', 'woocommerce' ),
```

claudiosanches commented on i18n/states.php:
```diff
@@ -172,6 +172,20 @@
+		'BJ-01' => __( 'Alibori', 'woocommerce' ),
+		'BJ-02' => __( 'Atacora', 'woocommerce' ),
+		'BJ-03' => __( 'Atlantique', 'woocommerce' ),
+		'BJ-04' => __( 'Borgou', 'woocommerce' ),
```
```suggestion
		'BO' => __( 'Borgou', 'woocommerce' ),
```

claudiosanches commented on i18n/states.php:
```diff
@@ -172,6 +172,20 @@
+		'BJ-02' => __( 'Atacora', 'woocommerce' ),
+		'BJ-03' => __( 'Atlantique', 'woocommerce' ),
+		'BJ-04' => __( 'Borgou', 'woocommerce' ),
+		'BJ-05' => __( 'Collines', 'woocommerce' ),
```
```suggestion
		'CO' => __( 'Collines', 'woocommerce' ),
```

claudiosanches commented on i18n/states.php:
```diff
@@ -172,6 +172,20 @@
+		'BJ-04' => __( 'Borgou', 'woocommerce' ),
+		'BJ-05' => __( 'Collines', 'woocommerce' ),
+		'BJ-06' => __( 'Couffo', 'woocommerce' ),
+		'BJ-07' => __( 'Donga', 'woocommerce' ),
```
```suggestion
		'DO' => __( 'Donga', 'woocommerce' ),
```

claudiosanches commented on i18n/states.php:
```diff
@@ -172,6 +172,20 @@
+		'BJ-05' => __( 'Collines', 'woocommerce' ),
+		'BJ-06' => __( 'Couffo', 'woocommerce' ),
+		'BJ-07' => __( 'Donga', 'woocommerce' ),
+		'BJ-08' => __( 'Littoral', 'woocommerce' ),
```
```suggestion
		'LI' => __( 'Littoral', 'woocommerce' ),
```

claudiosanches commented on i18n/states.php:
```diff
@@ -172,6 +172,20 @@
+		'BJ-06' => __( 'Couffo', 'woocommerce' ),
+		'BJ-07' => __( 'Donga', 'woocommerce' ),
+		'BJ-08' => __( 'Littoral', 'woocommerce' ),
+		'BJ-09' => __( 'Mono', 'woocommerce' ),
```
```suggestion
		'MO' => __( 'Mono', 'woocommerce' ),
```

claudiosanches commented on i18n/states.php:
```diff
@@ -172,6 +172,20 @@
+		'BJ-07' => __( 'Donga', 'woocommerce' ),
+		'BJ-08' => __( 'Littoral', 'woocommerce' ),
+		'BJ-09' => __( 'Mono', 'woocommerce' ),
+		'BJ-10' => __( 'Oueme', 'woocommerce' ),
```
```suggestion
		'OU' => __( 'Ouémé', 'woocommerce' ),
```

claudiosanches commented on i18n/states.php:
```diff
@@ -172,6 +172,20 @@
+		'BJ-08' => __( 'Littoral', 'woocommerce' ),
+		'BJ-09' => __( 'Mono', 'woocommerce' ),
+		'BJ-10' => __( 'Oueme', 'woocommerce' ),
+		'BJ-11' => __( 'Plateau', 'woocommerce' ),
```
```suggestion
		'PL' => __( 'Plateau', 'woocommerce' ),
```

claudiosanches commented on i18n/states.php:
```diff
@@ -172,6 +172,20 @@
+		'BJ-09' => __( 'Mono', 'woocommerce' ),
+		'BJ-10' => __( 'Oueme', 'woocommerce' ),
+		'BJ-11' => __( 'Plateau', 'woocommerce' ),
+		'BJ-12' => __( 'Zou', 'woocommerce' ),
```
```suggestion
		'ZO' => __( 'Zou', 'woocommerce' ),
```

claudiosanches reviewed (dismissed):
Looks good, but we also need to follow the code from the CLDR.

rodrigoprimo committed:
Use codes and names from CLDR for Benin subdivisions

Source: https://github.com/unicode-org/cldr/blob/master/common/subdivisions/en.xml#L521

Co-authored-by: Claudio Sanches <contato@claudiosanches.com>

rodrigoprimo reviewed (approved):
I approved the suggestions from Claudio to use codes from CLDR. This PR looks good to me now.
